staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -201,26 +285,116 @@ func (q *delayingType) ShutDown() {
 	})
 }
 
-// AddAfter adds the given item to the work queue after the given delay
+// Add follows the same behaviour as Add() on Interface(), except that it also
+// removes a duplicate item from the 'waiting' queue, if it exists.
+// All calls to Add are synchronous to preserve backwards compatibility with existing clients.
```

Comment:
Since you can pass your own queue via DelayingQueueConfig, I think this becomes quite problematic. Not only will we be overriding the behavior of our implementation, but we will also be overriding the behavior of custom queues that users may depend on.

Even in a case where our queue is used, people may have references both the `Type` and the `delayingType` and expect the same behaviour. But it will be different depending on which reference gets called.

For posterity, linking the other known problems: https://github.com/kubernetes/kubernetes/pull/112328/files#r1025739480.

I think, the consequences of this change are not fully foreseeable and it would be better to wait with such a change.

Reply:
Ok, making Add and AddAfter behave exactly the same as before.
Now to get the new behaviour you must call the new methods.

Reply:
Great, although AddAfter will still differ due to `PermitActiveAndWaiting` being false by default. What do you think about making the original behaviour the default and doing deduplication on demand? If you prefer to keep the deduplication, I can make a patch for our controllers to allow the duplication as it can cause races in there. Probably would be good to merge it together with this PR.

Reply:
Thanks for updating the behaviour of the `AddAfter` to match the original.